### ISSUE TITLE
Beta - Polish Translation Progress

### ISSUE THREAD

**Issue by jakubdrozdek** [reactjs/pl.reactjs.org#287]
# Beta Translation Progress
## Original issue
This issue was created on: https://github.com/reactjs/reactjs.org/issues/4135

## Progress
- [ ] ✅ Installation (note: a lot of content here is similar to old website)
  - [ ] ✅ Start a New React Project
  - [ ] ✅ Add React to a Website
  - [ ] ✅ Editor Setup
  - [ ] ✅ React Developer Tools
- [ ] 🔴 Quick Start (don't translate the index page yet, it will change)
  - [ ] ✅ Thinking in React
- [ ] 🔴 Describing the UI (don't translate the index page yet, it might change)
  - [ ] ✅ Your First Component
  - [ ] ✅ Importing and Exporting Components
  - [ ] ✅ Writing Markup with JSX
  - [ ] ✅ JavaScript in JSX with Curly Braces
  - [ ] ✅ Passing Props to a Component
  - [ ] ✅ Conditional Rendering
  - [ ] ✅ Rendering Lists
  - [ ] ✅ Keeping Components Pure
- [ ] 🔴 Adding Interactivity (don't translate the index page yet, it might change)
  - [ ] ✅ Responding to Events
  - [ ] ✅ State: A Component's Memory
  - [ ] ✅ Render and Commit
  - [ ] ✅ State as a Snapshot
  - [ ] ✅ Queueing a Series of State Updates
  - [ ] ✅ Updating Objects in State
  - [ ] ✅ Updating Arrays in State
- [ ] 🔴 Managing State (don't translate the index page yet, it might change)
  - [ ] ✅ Reacting to Input with State
  - [ ] ✅ Choosing the State Structure
  - [ ] ✅ Sharing State Between Components
  - [ ] ✅ Preserving and Resetting State
  - [ ] ✅ Extracting State Logic into a Reducer
  - [ ] ✅ Passing Data Deeply with Context
  - [ ] ✅ Scaling Up with Reducer and Context
- [ ] 🔴 Escape Hatches (not written yet)
  - [ ] ✅ Referencing Values with Refs
  - [ ] ✅ Manipulating the DOM with Refs

## Translating
If you want to help with translation, please leave a comment in this issue and write the name of the page you would like to translate. One of the repo maintainers will add your GitHub nickname next to the page to "make a reservation".

**Comment by lukbar48** [comment]
I will take Conditional rendering

**Comment by cyntler** [comment]
@lukbar48 do you do something with `Rendering Lists`?

**Comment by cyntler** [comment]
@jakubdrozdek I can take `Keeping Components Pure`. Could you assign it to me?

**Comment by gaearon** [comment]
Hi folks,

Just a heads-up that we've done our best to bring `main` and `legacy` branches up-to-date with upstream. This brings you to the latest visual design and sets up all the redirects. We don't anticipate more changes necessary on `legacy`. For `main`, there's a few things you need to be aware of:

- Keep in mind translating UI is best done closer to the end. We plan to make non-trivial code changes to the repo over the next few months. So it's best to focus on the content rather than on UI components.
- The homepage is untranslated so ideally it would be good if you could do a pass at that. For now, unfortunately it requires messing with the code file.
- A few rewritten pages (from Installation) are currently untranslated.
- This was a massive PR to merge, so we might have regressed somewhere or missed some changes that needed to be translated. Ideally, it would be great if you could go over the already-translated pages at your own pace and verify nothing important got lost.

Thank you so much for your cooperation!

**Comment by jakub-drozdek-noumena** [comment]
Hey @gaearon , I'll take a look at this soon and try to update our translations for the new docs. Thanks for resolving all conflicts. I saw how big and complex they were. I'm glad you did this for us 😅

**Comment by jakubdrozdek** [comment]
Hey @gaearon , I'll take a look at this soon and try to update our translations for the new docs. Thanks for resolving all conflicts. I saw how big and complex they were. I'm glad you did this for us 😅

**Comment by gaearon** [comment]
Ah this wasn't me, that's all @rickhanlonii's doing :)

**Comment by gaearon** [comment]
I've updated the template to the current page list.

**Comment by 0xaddm** [comment]
Hey, I could pick up the "Start a New React Project"

**Comment by dawidowski** [comment]
Hi @jakubdrozdek, I would like to contribute a bit. I can start with `useCallback`.

**Comment by dawidowski** [comment]
PR for `useCallback` is ready. I would like to start with `useEffect` now.

**Comment by dawidowski** [comment]
I would like to translate "Start a New React Project" section.

**Comment by dawidowski** [comment]
I would like to take "Rendering Lists" from "Describing the UI".

**Comment by tomaszwysocki** [comment]
I would like to translate "Add React to an Existing Project" page.

**Comment by dawidowski** [comment]
@jakubdrozdek, I would like to. take now _Keeping Components Pure_ .

Two PRs already wait for for next comments or approval ;)
#444 - _Start a New React Project_
#460 - _Rendering Lists_

**Comment by dawidowski** [comment]
I would like to take _Render and Commit_ page - need something light now ;)

**Comment by wizarddos** [comment]
Can I pick up _Add react to an existing project_ page?

**Comment by tomaszwysocki** [comment]
@wizarddos  I was doing that. It has been pushed already.

**Comment by wizarddos** [comment]
@tomaszwysocki Oh, all right

**Comment by dawidowski** [comment]
I would like to  [Adding Interactivity](https://pl.react.dev/learn/adding-interactivity) now.

**Comment by dawidowski** [comment]
PR for [Adding Interactivity](https://pl.react.dev/learn/adding-interactivity) is ready under #478.
I would like to translate [Preserving and Reseting State](https://pl.react.dev/learn/preserving-and-resetting-state) now.

Moreover, I see [Add React to an Existing Project](https://pl.react.dev/learn/add-react-to-an-existing-project) is already translated by @tomaszwysocki but not marked as done in the checklist. @jakubdrozdek, could you update it? ;)

**Comment by JohnyKam** [comment]
Hi. I would like to translate _Tutorial: Tic Tac Toe_ page. I hope that I will handle it. :-)

**Comment by wizarddos** [comment]
Can I take Responding to Events or State: A Component's Memory ?

**Comment by dawidsabat** [comment]
@JohnyKam and @wizarddos, welcome aboard! Good to see we have more contributors! 😄 

Basically, you can take whatever is not taken yet. Posting here what you want to translate is enough.
Take a look into [PL Consultations](https://github.com/reactjs/pl.react.dev/issues/3) and [TRANSLATIONS.md](https://github.com/reactjs/pl.react.dev/blob/main/TRANSLATION.md) to follow patterns and glossary. Feel free to propose new translations!

@JohnyKam, I would recommend to start with some shorter page. [Tutorial: Tic Tac Toe](https://pl.react.dev/learn/tutorial-tic-tac-toe) is quite big page to start with.
But if you fill you will handle it, go for it 😃 

By the way, @jakubdrozdek, I have recently changed my nickname from @dawidowski to @dawidsabat. Could I ask you to change it as well in the starting post above? I would love to do it myself but you are the author of this post

**Comment by JohnyKam** [comment]
True, _Tutorial: Tic Tac Toe_ is quite bit. Can I start with [Sharing State Between Components](https://pl.react.dev/learn/sharing-state-between-components)? Then, if necessary, I will move on to [Tutorial: Tic Tac Toe](https://pl.react.dev/learn/tutorial-tic-tac-toe) :-).

**Comment by dawidsabat** [comment]
@JohnyKam, go for it! 😀

**Comment by dawidsabat** [comment]
I would like to translate [State as a Snapshot](https://pl.react.dev/learn/state-as-a-snapshot) now.

@jakubdrozdek, @wizarddos - PR #490 for [Preserving and Resetting State](https://pl.react.dev/learn/preserving-and-resetting-state) is ready for the review. I would love to hear your feedback 😉

**Comment by Rohhan415** [comment]
May I translate Responding to Events?

**Comment by dawidsabat** [comment]
Hi @Rohhan415! @wizarddos took it and almost finished it in #491. 
What about [State: a component memory](https://pl.react.dev/learn/state-a-components-memory)? 😃 

@jakubdrozdek, the board is out of date. Can you mark _Responding to Events_ as in progress by @wizarddos and _State as a snapshot_ as done by me? Thanks!

**Comment by Rohhan415** [comment]
Sure, it doesn't matter which one. I’m just happy to help :)

**Comment by jakubdrozdek** [comment]
@dawidsabat Hey, I’d appreciate it if you could handle this. You’ve been a great contributor to this repository, and your translation work is solid, so I trust you to take on more responsibility. Let me know if you need any help!

I’m currently tied up with some personal matters and won’t be able to dedicate extra time to translations for now.

**Comment by dawidsabat** [comment]
Thank you for your trust @jakubdrozdek! I will do my best! 😄 🚀

**Comment by piotr-pajak** [comment]
Could I contribute by translating something from the "Components" section, for example, the `<Fragment> (<>)` or `<Profiler>` pages?

**Comment by dawidsabat** [comment]
Hi @piotr-pajak! Glad to see new face 😃 

In general, I would recommend you to select something closer to the top of the list, but if you really want, you can start with translation of something from the Components. No problem.

So, what do you choose?

**Comment by piotr-pajak** [comment]
Hi @dawidsabat! Thanks for the suggestion!
If it's best to start with something higher on the list, I'd be happy to work on Escape Hatches.
I'd like to start with Referencing Values with Refs 🚀

**Comment by dawidsabat** [comment]
@piotr-pajak, go for it, it's yours!

**Comment by Rohhan415** [comment]
Can I start translating 'Queueing a Series of State Updates'?
